dune/stuff/grid/periodicview.hh

```diff
@@ -0,0 +1,531 @@
+//
+// Contributors: Tobias Leibner
+
+#ifndef DUNE_STUFF_GRID_PERIODICVIEW_HH
+#define DUNE_STUFF_GRID_PERIODICVIEW_HH
+
+#include <bitset>
+#include <map>
+#include <utility>
+#include <vector>
+
+#include <dune/grid/common/gridview.hh>
+
+#include <dune/stuff/common/exceptions.hh>
+#include <dune/stuff/common/memory.hh>
+#include <dune/stuff/common/float_cmp.hh>
+#include <dune/stuff/grid/search.hh>
+
+namespace Dune {
+namespace Stuff {
+namespace Grid {
+
+
+template< class RealGridViewImp >
+class PeriodicIntersection
+    : public RealGridViewImp::Intersection
+{
+  typedef RealGridViewImp                          RealGridViewType;
+  typedef PeriodicIntersection< RealGridViewType > ThisType;
+  typedef typename RealGridViewType::Intersection  BaseType;
+
+public:
+  typedef typename BaseType::LocalGeometry                LocalGeometry;
+  typedef typename BaseType::Geometry::GlobalCoordinate   GlobalCoordinate;
+  typedef typename BaseType::EntityPointer                EntityPointer;
+  typedef typename RealGridViewType::IntersectionIterator RealIntersectionIteratorType;
+  static const size_t dimDomain = RealGridViewType::dimension;
+
+  //! \brief Constructor from real intersection
+  PeriodicIntersection(const BaseType& real_intersection, const RealGridViewType& real_grid_view,
+                       const std::pair< bool, EntityPointer >& periodic_pair)
+    : BaseType(real_intersection)
+    , periodic_(periodic_pair.first)
+    , outside_(periodic_pair.second)
+    , real_grid_view_(&real_grid_view)
+  {}
+
+  //! \brief Copy operator from existing PeriodicIntersection
+  const ThisType operator=(ThisType other)
+  {
+    real = other.real;
+    periodic_ = other.periodic_;
+    outside_ = other.outside_;
+    real_grid_view_ = other.real_grid_view_;
+    return *this;
+  }
+
+  // methods that differ from BaseType
+  bool neighbor() const
+  {
+    if (periodic_)
+      return true;
+    else
+      return BaseType::neighbor();
+  }
+
+  bool boundary() const
+  {
+    if (periodic_)
+      return false;
+    else
+      return BaseType::boundary();
+  }
+
+  bool periodic() const
+  {
+    return periodic_;
+  }
+
+  EntityPointer outside() const
+  {
+    if (periodic_)
+      return outside_;
+    else
+      return BaseType::outside();
+  }
+
+  LocalGeometry geometryInOutside() const
+  {
+    if (periodic_) {
+      const BaseType outside_intersection = find_intersection_in_outside();
+      return outside_intersection.geometryInInside();
+    } else {
+      return BaseType::geometryInOutside();
+    }
+  } //geometryInOutSide
+
+  int indexInOutside() const
+  {
+    if (periodic_) {
+      const BaseType outside_intersection = find_intersection_in_outside();
+      return outside_intersection.indexInInside();
+    } else {
+      return BaseType::indexInOutside();
+    }
+  }
+
+private:
+  // tries to find intersection in outside (works only if periodic_ == true)
+  const BaseType find_intersection_in_outside() const {
```

Comment:
This does not compile for me as well (neither clang 3.5.1 nor gcc 4.9.2), since the `BaseType` intersection does not allow copy.